astroquery/esa/jwst/core.py

```diff
@@ -978,6 +979,8 @@ def get_obs_products(self, *, observation_id=None, cal_level="ALL",
             Returns the local path where the product(s) are saved.
         """
 
+        if (type(product_type) is list and '' in product_type) or not product_type:
```

Comment:
isinstance(product_type, list) is usually cleaner than what is there, but this is nitpicking. The bigger issue is that this will be True for a list that looks like: `['', 'science', 'info']` which I suppose is not the intended behaviour.

Howeverk duck-typing is real, and the documentation above is clear, so dropping these two lines should be enough. If the user provides a string, or a list that only contains `''`, it's on to them, the docs lists all the possible values.

Reply:
With empty element, in the list or in the string, the code will download all products (product_type = None). In your example: ['', 'science', 'info'] is a malformed list so, all products will be downloaded.
We can change "type + is" check to isinstance, no problem